pkg/auth/auth_test.go

```diff
@@ -202,3 +205,59 @@ func Test_GetCertRenewRequiredDelay(t *testing.T) {
 		t.Errorf("RenewRequired Expected:true Actual:false")
 	}
 }
+
+func Test_CertCheckNotExpired(t *testing.T) {
+	now := time.Now().UTC()
+
+	tmpl := x509.Certificate{
+		SerialNumber: new(big.Int).SetInt64(0),
+		Subject: pkix.Name{
+			CommonName: "test",
+		},
+		NotBefore: now,
+		NotAfter:  now.Add(time.Second * 10),
+	}
+
+	b, err := x509.CreateCertificate(rand.Reader, &tmpl, &tmpl, key.Public(), key)
+	if err != nil {
+		t.Errorf("Failed creating certificate %v", err)
+	}
+
+	x509Cert, err := x509.ParseCertificate(b)
+	if err != nil {
+		t.Errorf("Failed parsing certificate %v", err)
+	}
+
+	certPem := certs.EncodeCertPEM(x509Cert)
+	if err = certCheck(certPem); err != nil {
+		t.Errorf("certCheck Expected:nil Actual:%v", err)
+	}
+}
+
+func Test_CertCheckExpired(t *testing.T) {
+	now := time.Now().UTC()
+
+	tmpl := x509.Certificate{
+		SerialNumber: new(big.Int).SetInt64(0),
+		Subject: pkix.Name{
+			CommonName: "test",
+		},
+		NotBefore: now.Add(time.Second * -10),
+		NotAfter:  now.Add(time.Second * -1),
+	}
+
+	b, err := x509.CreateCertificate(rand.Reader, &tmpl, &tmpl, key.Public(), key)
+	if err != nil {
+		t.Errorf("Failed creating certificate %v", err)
+	}
+
+	x509Cert, err := x509.ParseCertificate(b)
+	if err != nil {
+		t.Errorf("Failed parsing certificate %v", err)
+	}
+
+	certPem := certs.EncodeCertPEM(x509Cert)
+	if err = certCheck(certPem); err == nil || !errors.IsExpired(err) {
```

Comment:
Should these be 2 separate checks?
Will certCheck() return an error is the cert is expired?
If so, you will never evaluate isExpired() and it will never be tested.

Reply:
Oh, yeah, now I see the certCheck() function :)
you should have 2 separate checks.
1. check that certCheck() returns expire cert.  It can return an error for other reasons
2. check that isExpired returns true.